### DISCUSSION TITLE
Next features francis_V1

### DISCUSSION THREAD

**Discussion by Hboni** [Hboni/francis#39] [Ideas]
With the first version of francis, we can now think about features that we want to integrate, I put some ideas here:
1. Add Max/Min operations between images
2. Add some morphology operations like closing/opening
3. Be possible to save pipeline
4. A button to update all the nodes?
5. Use of values from images (like get the max of an image and use this value to another node)
6. Be possible to make threshold in percentage
7. Manage other types of image than .nii.gz

This is some features that I though would be great to have, do you have other ideas?

**Comment by jeaunrg** [comment]
A few questions:
1. you want a process that return an array with same size of all input images, with local min/max between images ? Or just a global min/max value?
7. which types exactly?

**Reply by Hboni** [reply]
1. I first thought only a value for global max/min. One idea is the possibility to thresh an image according to the max of another image. It may need development to manage int and not only images.
2. The most interesting for me is opening and closing, to avoid using erosion and dilation. But we can think about watershed also. If you have others in mind it can be great also.

Do you have other ideas? Which ones of those features do you think we can reasonably add in a next release?

**Reply by jeaunrg** [reply]
It is really simple to add those features (in fact opening and closing are already in basic morpho oups).
But with more complex features it is necessary to put processes in a thread, else the app will freeze for a very long time. 
So I think we should add a Threading decorator on module function before anything else..

**Reply by jeaunrg** [reply]
and 7. which types of image besides .nii.gz?

**Reply by Hboni** [reply]
Ok, I see, you can put the threading issue inside the Francis_1.1 project. You can also complete the list with features we want to integrate for the next release in the actual note, to have an overview of what we want to do!

7. I thought about png or jpeg to simple images, maybe tiff. In case of development for it, I will put example images in the project, for test purpose

**Reply by jeaunrg** [reply]
ok thanks !
